fedimint-bin-tests/src/federation.rs

```diff
@@ -0,0 +1,320 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use fedimint_core::core::LEGACY_HARDCODED_INSTANCE_ID_WALLET;
+use fedimint_core::db::mem_impl::MemDatabase;
+use fedimint_core::task::TaskGroup;
+use fedimint_wallet_client::config::WalletClientConfig;
+use tokio::fs;
+use tokio::sync::mpsc::{self, Receiver, Sender};
+
+use super::*; // TODO: remove this
+
+/// Create a map of environment variables which fedimintd and DKG can use,
+/// but which can't be defined by `build.sh` because multiple of these daemons
+/// run concurrently with different values.
+///
+/// We allow ranges of 10 ports for each fedimintd / dkg instance starting from
+/// 18173. Each port needed is incremented by 1 within this range.
+///
+/// * `id` - ID of the server. Used to calculate port numbers.
+fn fedimint_env(id: usize) -> anyhow::Result<HashMap<String, String>> {
```

Comment:
No need to change it now, but `id: usize` is vague. Could we use `PeerId`? Or is it just a sequence `peer_i : usize`?

Reply:
Didn't use `PeerId` because it is just less ergonomic.